src/btree/row_key.c

```diff
@@ -315,13 +315,6 @@ __wt_row_leaf_key_work(
          * prefix compressed.
          */
         if (unpack->prefix == 0) {
-            /*
-             * The only reason to be here is a Huffman encoded key, a non-encoded key with no prefix
-             * compression should have been directly referenced, and we should not have needed to
-             * unpack its cell.
-             */
-            WT_ASSERT(session, btree->huffman_key != NULL);
```

Comment:
Why this assert is removed? Looks like irrelevant to the change.

Reply:
This code path is hit when we don't encode the location into the row. Previously this only happens when we have a key that we're compressing with. Now we've added a stricter condition that takes into account the visibility of the unpacked value so this no longer holds.